### ISSUE TITLE
ldeinit will take segmentation fault if given keyboard/mouse input

### ISSUE THREAD

**Issue by nbriggs** [Interlisp/medley#792]
The code in the "INIT" configuration defers initializing the keyboard, cursor, and mouse handling code because it caused a problem when running on a serial-connected keyboard device (e.g., Sun workstation) and the Lisp code was not prepared to handle the I/O.  Under X11 (or, presumably, SDL) the code attempts to process the X events and fails because of the lack of initialization.   A simple fix is to not process X events in the INIT configuration -- no interactive execution is expected or desired at this stage.

**Comment by rmkaplan** [comment]
Is this the cause of seemingly random faults during a mkinit—if you happen to type something during a critical stage?

**Comment by nbriggs** [comment]
Yes.   If you move the mouse over the window or give it the input focus or click or type… anything that processes X events will cause it to fail like this.

— Nick

**Comment by nbriggs** [comment]
It used to fail in the `loadup-mid-from-init.sh` script (where it uses ldeinit) -- the way to test this is, if you've got a previous loadup sitting there, just run `./scripts/loadup-mid-from-init.sh` and while its X window is up, click and move the mouse over that window.   It used to fail every time you did that but with the new one it should never fail.

**Comment by masinter** [comment]
I suppose it should be an open issue to clean up the loadup process and (in particular) get rid of the need for ldeinit.

**Comment by nbriggs** [comment]
I suspect it still needs "init" mode (even if it doesn't have a separate executable) because we can't have it processing keyboard or mouse/cursor events (from the hard keyboard, or X/SDL events) and putting that state into the newly created sysout, or referencing Lisp code/atoms that may not exist in the sysout at that stage -- such as `SI:*CLOSURE-CACHE*`, `IL:\TEDIT.BLTCHAR`, `IL:\PUNT.BITBLT.BITMAP` and so on (perhaps anything that has a package before the package system is loaded)
